runtime/src/account_storage.rs

```diff
@@ -92,8 +92,14 @@ impl AccountStorage {
 
     /// remove the append vec at 'slot'
     /// returns the current contents
-    pub(crate) fn remove(&self, slot: &Slot) -> Option<Arc<AccountStorageEntry>> {
-        assert!(self.shrink_in_progress_map.is_empty());
+    pub(crate) fn remove(
+        &self,
+        slot: &Slot,
+        shrink_can_be_active: bool,
+    ) -> Option<Arc<AccountStorageEntry>> {
+        if !shrink_can_be_active {
+            assert!(self.shrink_in_progress_map.is_empty());
+        }
```

Comment:
(Optional and totally up to you)

Maybe slightly more concise - either the shrink can be in progress or it can be empty (or both).
```suggestion
        assert!(shrink_can_be_active || self.shrink_in_progress_map.is_empty());
```

Reply:
I did this. I had to fix rebase conflicts anyway.